src/components/PetitionsFilters/index.js

```diff
@@ -1,32 +1,147 @@
 import React from 'react';
 import settings from 'settings';
+import { petitionsPath } from 'helpers/petitionUrls';
+import citySuggestionFormatter from 'helpers/citySuggestionFormatter';
 import Icon from 'components/Icon';
 import Autocomplete from 'containers/Autocomplete';
 import styles from './petitions-filters.scss';
 
-const PetitionsFilters = ({ autocompleteProps }) => (
-  <div className={styles.root}>
-    <div className={styles.filter}>
-      <label className={styles.label} htmlFor={autocompleteProps.name}>
-        {settings.petitionsPage.filters.city.label}
-      </label>
-
-      <div className={styles['input-wrapper']}>
-        <Autocomplete
-          {...autocompleteProps}
-          inputModifier={'thin'}
-        />
-
-        <div className={styles.icon}>
-          <Icon
-            id={'Search'}
-            inline
-            fill={'none'}
-          />
+const FILTER_INPUT_NAME = 'filter-by';
+const SORT_INPUT_NAME = 'sort-by';
+const CITY_FILTER_NAME = 'city-filter';
+
+const PetitionsFilters = React.createClass({
+  handleFilterChange (e) {
+    this.props.router.push(petitionsPath({
+      state: e.target.value,
+      city: this.props.currentCity || '',
+      sort: this.props.location.query.sort || ''
+    }));
+  },
+
+  handleSortChange (e) {
+    this.props.router.push(petitionsPath({
+      state: this.props.params && this.props.params.state || this.props.location.query.state || '',
+      city: this.props.currentCity || '',
+      sort: e.target.value
+    }));
+  },
+
+  getAutocompleteProps: ({
+    router,
+    params,
+    location,
+    updateCurrentCity,
+    currentCity
+  }) => ({
+    name: CITY_FILTER_NAME,
+    endpoint: 'cities',
+    suggestionFormatter: citySuggestionFormatter,
+    getFormValue: (suggestion) => suggestion,
+    suggestionsLimit: 4,
+    helper: {
+      value: { data: currentCity },
+      onChange (newValue) {
+        updateCurrentCity(newValue);
+
+        router.push(petitionsPath({
+          state: params && params.state || location.query.state || '',
+          city: newValue.id ? newValue : '',
+          sort: location.query.sort || ''
+        }));
+      },
+      onBlur () {}
+    },
+    html: { placeholder: settings.petitionsPage.filters.city.placeholder }
+  }),
+
+  render () {
```

Comment:
I think this component is a little big and complex. Any way we can break it down?